### PR TITLE
Improve elixir hooks

### PR DESCRIPTION
**Pull request by greenfork**
Improvements:
- Elixir has tests now
- Auto-insert "end" keyword
- Proper comment prefix copying `#`

"end" auto-insert is partially copied from Ruby

### CODE REVIEW

File: rc/filetype/elixir.kak [hunk 1]
```diff
@@ -92,12 +92,25 @@ define-command -hidden elixir-trim-indent %{
...
-define-command -hidden elixir-insert-on-new-line %{
-    evaluate-commands -draft -itersel %{
-        # copy -- comments prefix and following white spaces
-        try %{ execute-keys -draft k <a-x> s ^\h*\K--\h* <ret> y gh j P }
-    }
-}
+define-command -hidden elixir-insert-on-new-line %[
+    evaluate-commands -no-hooks -draft -itersel %[
+        # copy '#' comment prefix and following white spaces
+        try %{ exec -draft k <a-x> s ^\h*#\h* <ret> y jgh P }
```
Comment (krobelus) [inline rc/filetype/elixir.kak:98]: This creates wrong indentation for indented comments (like #1118). Test case:
```
def test do
    # comment%()
end
```
`c<ret>` will add spurious indentation to the new line.

Going back to the old regex with `\K` fixes it.

**Review by krobelus — CHANGES_REQUESTED** [review]
Looks good - just one thing

File: rc/filetype/elixir.kak (hunk 1 above)
Comment (greenfork) [reply rc/filetype/elixir.kak]: Thanks a lot for pointing this out. I fixed it, added tests to cover that case and did similar fixes to adjacent Ruby PR https://github.com/mawww/kakoune/pull/4217

File: rc/filetype/elixir.kak [hunk 2]
```diff
@@ -92,12 +92,25 @@ define-command -hidden elixir-trim-indent %{
...
-define-command -hidden elixir-insert-on-new-line %{
-    evaluate-commands -draft -itersel %{
-        # copy -- comments prefix and following white spaces
-        try %{ execute-keys -draft k <a-x> s ^\h*\K--\h* <ret> y gh j P }
-    }
-}
+define-command -hidden elixir-insert-on-new-line %[
+    evaluate-commands -no-hooks -draft -itersel %[
+        # copy '#' comment prefix and following white spaces
+        try %{ exec -draft k <a-x> s ^\h*\K#\h* <ret> y jgi P }
```
Comment (krobelus) [inline rc/filetype/elixir.kak:98]:
```suggestion
        try %{ execute-keys -draft k <a-x> s ^\h*\K#\h* <ret> y jgi P }
```

for consistency with other rc files, same for the ruby PR

Comment (greenfork) [reply rc/filetype/elixir.kak]: Sure, done

**Pull request merged**